script/dialog/roll-dialog.js

```
@@ -0,0 +1,135 @@
+import DiceRoller from "../components/dice-roller.js";
+
+export class RollDialog {
+    
+    /**
+     * Display roll dialog and execute the roll.
+     * 
+     * @param  {string}        rollName
+     * @param  {object|number} baseDefault     {name: "somename", value: 5} | 5
+     * @param  {object|number} skillDefault    {name: "somename", value: 5} | 5
+     * @param  {number}        gearDefault
+     * @param  {string}        artifactDefault
+     * @param  {number}        modifierDefault
+     * @param  {number}        damage
+     * @param  {DiceRoller}    [diceRoller]
+     * @param  {callback}      [onAfterRoll]
+     */
+    static prepareRollDialog(rollName, baseDefault, skillDefault, gearDefault, artifactDefault, modifierDefault, damage, diceRoller, onAfterRoll) {
+        diceRoller = diceRoller || new DiceRoller();
```

Making the diceRoller optional is not a good idea. It will be impossible to push this roll if you forget it. It's better to throw an exception so people will know that they need a diceRoller.
In fact, you forgot the diceRoller on some call but it's not visible until you try to push the roll.

---

I made it optional because not all rolls can be pushed, but I see your point. I'll make it required.